src/coreclr/jit/lclmorph.cpp

```diff
@@ -909,13 +911,45 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
 
             case IndirTransform::BitCast:
                 indir->ChangeOper(GT_BITCAST);
-                indir->gtGetOp1()->ChangeOper(GT_LCL_VAR);
-                indir->gtGetOp1()->ChangeType(varDsc->TypeGet());
-                indir->gtGetOp1()->AsLclVar()->SetLclNum(lclNum);
-                lclNode = indir->gtGetOp1()->AsLclVarCommon();
+                lclNode = BashToLclVar(indir->gtGetOp1(), lclNum);
                 break;
 
+#ifdef FEATURE_HW_INTRINSICS
+            case IndirTransform::GetElement:
+            {
+                var_types elementType = indir->TypeGet();
+                assert(elementType == TYP_FLOAT);
+
+                lclNode            = BashToLclVar(indir->gtGetOp1(), lclNum);
+                GenTree* indexNode = m_compiler->gtNewIconNode(val.Offset() / genTypeSize(elementType));
+                GenTree* hwiNode   = m_compiler->gtNewSimdGetElementNode(elementType, lclNode, indexNode,
+                                                                       CORINFO_TYPE_FLOAT, genTypeSize(varDsc),
+                                                                       /* isSimdAsHWIntrinsic */ false);
+                indir->ReplaceWith(hwiNode, m_compiler);
```

Comment:
Is should be possible in principle (we using local visitor after all), but it is not super trivial since so far we've been assuming all of the transformations bash nodes in place, and so upstream references to `node` remain "correct", and we don't need to thread the uses into `Value`s.

But it shouldn't be "hard" either, I will see what it looks like.

Reply:
`ReplaceWith` usage has been removed.